pkg/component/kubeapiserverexposure/kube_apiserver_sni_test.go

```diff
@@ -218,7 +218,7 @@ var _ = Describe("#SNI", func() {
 	})
 
 	Describe("#Deploy", func() {
-		It("succeeds", func() {
+		test := func() {
 			Expect(defaultDepWaiter.Deploy(ctx)).To(Succeed())
```

Comment:
I think we need `WithOffset` here now, or?

Reply:
I don't think that it's really valuable here because if `WithOffset` is used and the test fails, it points to the place in the code where `test()` is invoked instead of pointing to the assertion that led to the failing test result. The latter is rather something a dev is interested in?!

Reply:
Sure. However, do we need to adapt https://github.com/gardener/gardener/blob/master/docs/development/testing.md#writing-tests-relevant-for-all-kinds then to make the use cases more clear?

> Test failures should point to an exact location, so that failures in CI aren't too difficult to debug/fix.
> - Use `ExpectWithOffset` for making assertions in helper funcs like `expectSomethingWasCreated`: [example test](https://github.com/gardener/gardener/blob/2eb54485231408cbdbabaa49812572a07124364f/extensions/pkg/controller/controlplane/genericactuator/actuator_test.go#L732-L736)
> - Make sure to add additional descriptions to Gomega matchers if necessary (e.g. in a loop): [example test](https://github.com/gardener/gardener/blob/2eb54485231408cbdbabaa49812572a07124364f/test/e2e/shoot/internal/rotation/certificate_authorities.go#L89-L93)